extensions/control-plane/store/sql/contract-definition-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractdefinition/SqlContractDefinitionStore.java

```diff
@@ -62,7 +67,8 @@ public SqlContractDefinitionStore(DataSourceRegistry dataSourceRegistry, String
 
             try {
                 var queryStmt = statements.createQuery(spec);
-                return queryExecutor.query(getConnection(), true, this::mapResultSet, queryStmt.getQueryAsString(), queryStmt.getParameters());
+                return queryExecutor.query(getConnection(), true, this::mapContractDefinitionId, queryStmt.getQueryAsString(), queryStmt.getParameters())
+                        .map(this::findById);
```

Comment:
Doing two queries to get the contract definition seems overkill, the old behavior was ok

Reply:
This is addresses